Userland/Libraries/LibC/math.cpp

```diff
@@ -680,16 +717,34 @@ long double frexpl(long double x, int* exp) NOEXCEPT
     return scalbnl(x, -(*exp));
 }
 
-#if !(ARCH(X86_64))
-
-double round(double value) NOEXCEPT
+double round(double x) NOEXCEPT
```

Comment:
This pattern we're accumulating here of 

```
type func(type a, type b) {
#if ARCH(A)
    blah
#elif ARCH(B)
    blah
#elif ARCH(C)
   blah
#else
#    error "unknown arch"
#endif
}
```

Isn't very tasteful imo. A future PR should try to move these into arch-specific .cpp files when a common implementation is not possible or not desired.

Reply:
Indeed! I was following existing patterns here but IIRC there's faster implementations for most architectures for these rounding functions (even x86 probably has SSE hacks we can utilize). I will leave a FIXME.

Reply:
Not for `round`, but for `floor` and `trunk` SSE4.1 adds a dedicated instruction

Reply:
Since this is a concern across the project, I instead created an issue: #22657